hooks/useScrollPosition.tsx

```diff
@@ -0,0 +1,120 @@
+  const [maxScroll, setMaxScroll] = useState(0);
+  const [maxSize, setMaxSize] = useState(0);
+  const [position, setPosition] = useState(0);
+  const elementRef = useRef<ElementItems>(null);
+
+  // convert to our ref to track changes ... I think
+  const setElement = (element: HTMLElement) => {
+    // console.log("element set")
+    if (element?.scrollHeight) {
+      elementRef.current = {
+        ...elementRef.current,
+        maxScrollWidth: element.scrollHeight,
+      }
+    }
+    if (element?.scrollWidth) {
+      elementRef.current = {
+        ...elementRef.current,
+        maxScrollWidth: element.scrollWidth,
+      }
+    }
+    if (element?.clientHeight) {
+      elementRef.current = {
+        ...elementRef.current,
+        height: element.clientHeight,
+      }
+    }
+    if (element?.clientWidth) {
+      elementRef.current = {
+        ...elementRef.current,
+        width: element.clientWidth
+      }
+    }
+    // if scrolling horizontally
+    if (element?.scrollLeft) {
+      elementRef.current = {
+        ...elementRef.current,
+        horizontalScroll: element.scrollLeft,
+      }
+    }
+    // if scrolling vertically
+    if (element?.scrollTop) {
+      elementRef.current = {
+        ...elementRef.current,
+        verticalScroll: element.scrollTop,
+      }
+    }
+  }
+
+  // update maxScroll and position
+  const handleElementUpdates = () => {
+    if (elementRef.current === null) return;
+    // scrolling horizontally
+    if (elementRef.current?.horizontalScroll && !elementRef.current?.verticalScroll) {
+      const maxScrollWidth = elementRef.current?.maxScrollWidth - elementRef.current?.width;
+      setMaxScroll(maxScrollWidth || 0);
+      setPosition(elementRef.current?.horizontalScroll || 0);
+      setMaxSize(elementRef.current?.width || 0);
+    } 
+    // scrolling vertically
+    else if (elementRef.current?.verticalScroll && !elementRef.current?.horizontalScroll) {
+      const maxScrollHeight = elementRef.current?.maxScrollHeight - elementRef.current?.height;
+      setMaxScroll(maxScrollHeight || 0);
+      setPosition(elementRef.current?.verticalScroll || 0);
+      setMaxSize(elementRef.current?.height || 0);
+    }
+  }
+
+  /* 
+  I wanted to be able to update our exposed values whenever we update elementRef by listening to the changes to the ref to avoid having to call in
+  each eventListener. I thought useEffect would allow me to do this but it doesn't seem to work for some reason. Anybody know why? 
+  TODO: Fix or remove commented out code
+  */
+ // useEffect(() => {
+ //   handleElementUpdates();
+ // }, [elementRef.current]);
+
+
+  useEffect(() => {
+    const scrollContainer = document.getElementById(prop);
+    // do not proceed if you can't find the element
+    if (!scrollContainer) { return;} 
+    // set the element initial values
+    setElement(scrollContainer);
+
+    // TODO: handle more events
+    window.addEventListener('wheel', () => {
```

Comment:
if we add these event listeners to the scrollContainer, and change the first argument to 'scroll' it works for both mouse wheel and using the scroll bar that's sitting under the gallery

Reply:
Nice! I can move.

Reply:
Holy shit ... straight up spent so much time trying to figure out why when I changed wheel to scroll here that it wouldn't work 😢 . Great coms